### PR TITLE
adds participant 'address book'

### PR DESCRIPTION
**Pull request by hughy**
## Summary

adds a database store to the walletDb that stores identifiers for participants in the signing group of a multisig account

'participantIdentifiers' uses a compound key consisting of the account prefix and a participant identifier where each participant identifier is expected to be a hex string

implements methods for adding, deleting, and iterating over identifiers

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/wallet/walletdb/walletdb.ts [hunk 1]
```diff
@@ -274,6 +279,16 @@ export class WalletDB {
       valueEncoding: NULL_ENCODING,
     })
 
+    this.participantIdentifiers = this.db.addStore({
+      name: 'pa',
```
[the remark this answers is not in the document]
Comment (hughy) [reply ironfish/src/wallet/walletdb/walletdb.ts:283]: Ah, I renamed it from `participantAddresses` 😅

**Review by jowparks — APPROVED** [review]
LGTM, if you want you could put up a PR real fast to create a napi binding for Secret/Identifier and then use them in a test for storage:
https://github.com/iron-fish/ironfish-frost/blob/705f29d4010567bba1333912de0036a730387bff/src/participant.rs#L99-L104

You may need to reexport them to make them accessible in the napi layer.

**Comment by hughy** [comment]
> LGTM, if you want you could put up a PR real fast to create a napi binding for Secret/Identifier and then use them in a test for storage: https://github.com/iron-fish/ironfish-frost/blob/705f29d4010567bba1333912de0036a730387bff/src/participant.rs#L99-L104
> 
> You may need to reexport them to make them accessible in the napi layer.

When we move on to using the Secret/Identity from ironfish-frost in milestone 3 we'll need to store the Identity for each participant, but before then we'll only have the Identifier

I think the schema of this datastore may need to change to something like this:
```
{
  key: [account.prefix, Identifier],
  value: Identity | null
}
```

**Comment by jowparks** [comment]
> > LGTM, if you want you could put up a PR real fast to create a napi binding for Secret/Identifier and then use them in a test for storage: https://github.com/iron-fish/ironfish-frost/blob/705f29d4010567bba1333912de0036a730387bff/src/participant.rs#L99-L104
> > You may need to reexport them to make them accessible in the napi layer.
> 
> When we move on to using the Secret/Identity from ironfish-frost in milestone 3 we'll need to store the Identity for each participant, but before then we'll only have the Identifier
> 
> I think the schema of this datastore may need to change to something like this:
> 
> ```
> {
>   key: [account.prefix, Identifier],
>   value: Identity | null
> }
> ```

I am not sure we can go from `Identifier` back to `Secret`, so maybe we want to store Secret and derive identifier? Or store both?

**Comment by hughy** [comment]
Moved back to draft -- we won't need to track participant identifiers or identities in the walletDb until we use our own identifiers in Milestone 3.

File: ironfish/src/wallet/walletdb/walletdb.ts [hunk 2]
```diff
@@ -282,6 +288,16 @@ export class WalletDB {
       valueEncoding: NULL_ENCODING,
     })
 
+    this.participantIdentities = this.db.addStore({
+      name: 'pi',
+      keyEncoding: new PrefixEncoding(
+        new BufferEncoding(), // account prefix
+        new StringEncoding(), // participant identifier
```
Comment (hughy) [inline ironfish/src/wallet/walletdb/walletdb.ts:295]: maybe we change this to a Buffer? that will avoid the need to serialize into a string when we add an identity in `addParticipantIdentity` and make the key smaller

**Pull request merged**